lib/metadata/bucketfile/backend.js

```diff
@@ -0,0 +1,453 @@
+import net from 'net';
+import fs from 'fs';
+import cluster from 'cluster';
+import events from 'events';
+import assert from 'assert';
+
+import level from 'level';
+import multilevel from 'multilevel';
+import sublevel from 'level-sublevel';
+import arsenal from 'arsenal';
+
+import { logger } from '../../utilities/logger';
+import BucketInfo from '../BucketInfo';
+import constants from '../../../constants';
+import config from '../../Config';
+
+const errors = arsenal.errors;
+
+const METADATA_PORT = 9990;
+const METADATA_PATH = `${config.filePaths.metadataPath}/`;
+const MANIFEST_JSON = 'manifest.json';
+const MANIFEST_JSON_TMP = 'manifest.json.tmp';
+const ROOT_DB = 'rootDB';
+const METASTORE = '__metastore';
+const OPTIONS = { sync: true };
+
+class BucketFileInterface {
+
+    constructor() {
+        this.logger = logger;
+        if (cluster.isMaster) {
+            this.startServer();
+        } else {
+            this.refcnt = 0;
+            this.reConnect();
+            this.notifier = new events.EventEmitter();
+        }
+    }
+
+    /**
+     * Start the server
+     * @return {undefined}
+     */
+    startServer() {
+        const rootDB = level(METADATA_PATH + ROOT_DB);
+        const sub = sublevel(rootDB);
+        sub.methods = sub.methods || {};
+        sub.methods.createSub = { type: 'async' };
+        sub.createSub = (subName, cb) => {
+            try {
+                sub.sublevel(subName);
+                multilevel.writeManifest(sub,
+                                         METADATA_PATH +
+                                         MANIFEST_JSON_TMP);
+                fs.renameSync(METADATA_PATH + MANIFEST_JSON_TMP,
+                              METADATA_PATH + MANIFEST_JSON);
+                cb();
+            } catch (err) {
+                cb(err);
+            }
+        };
+        const metastore = sub.sublevel(METASTORE);
+        /* due to a chicken-and-egg issue we need to create
```

Comment:
Based on our discussion, I would change this to:

Since the bucket creation API is expecting the usersBucket to have attributes, we pre-create the usersBucket.